Mathlib/RingTheory/PolynomialLaw/Basic.lean

```diff
@@ -264,4 +279,345 @@ theorem id_comp : id.comp f = f := by ext; rfl
+    simp only [φ, aeval_X, Equiv.symm_apply_apply]
+
+/-- Extends `PolynomialLaw.isCompat` to all universes -/
+theorem isCompat {T : Type w} [CommSemiring T] [Algebra R T] (h : S →ₐ[R] T) :
+    h.toLinearMap.rTensor N ∘ f.toFun S = f.toFun T ∘ h.toLinearMap.rTensor M := by
+  ext t
+  simp only [Function.comp_apply, PolynomialLaw.isCompat_apply]
+
+end Lift
+
+section Module
+
+variable
+  {R : Type u} [CommSemiring R]
+  {M : Type*} [AddCommMonoid M] [Module R M]
+  {N : Type*} [AddCommMonoid N] [Module R N]
+  (r a b : R) (f g : M →ₚₗ[R] N)
+  {S : Type*} [CommSemiring S] [Algebra R S]
+
+/-- Extension of `PolynomialLaw.zero_def` -/
+@[simp]
+theorem toFun_zero : (0 : M →ₚₗ[R] N).toFun S = 0 := by
+  ext t
+  obtain ⟨⟨s, p⟩, ha⟩ := π_surjective t
+  simp only [toFun_eq_rTensor_φ_toFun' _ ha, zero_def, Pi.zero_apply, _root_.map_zero]
+
+/-- Extension of `PolynomialLaw.add_def_apply` -/
+@[simp]
+theorem add_toFun_apply (t : S ⊗[R] M) :
+    (f + g).toFun S t = f.toFun S t + g.toFun S t := by
+  obtain ⟨⟨s, p⟩, ha⟩ := π_surjective t
+  simp only [Pi.add_apply, toFun_eq_rTensor_φ_toFun' _ ha, add_def, map_add]
+
+/-- Extension of `PolynomialLaw.add_def` -/
+@[simp]
+theorem add_toFun :
+    (f + g).toFun S = f.toFun S + g.toFun S := by
+  ext t
+  simp only [Pi.add_apply, add_toFun_apply]
+
+@[simp]
+theorem neg_toFun {R : Type u} [CommRing R]
+    {M : Type*} [AddCommGroup M] [Module R M]
+    {N : Type*} [AddCommGroup N] [Module R N]
+    (f : M →ₚₗ[R] N)
+    (S : Type*) [CommSemiring S] [Algebra R S] :
+    (-f).toFun S = (-1 : R) • (f.toFun S) := by
+  ext t
+  obtain ⟨⟨s, p⟩, ha⟩ := π_surjective t
+  simp only [toFun_eq_rTensor_φ_toFun' _ ha, neg_def, Pi.smul_apply, map_smul]
+
+variable (S) in
+/-- Extension of `PolynomialLaw.smul_def` -/
+@[simp]
+theorem smul_toFun : (r • f).toFun S = r • (f.toFun S) := by
+  ext t
+  obtain ⟨⟨s, p⟩, ha⟩ := π_surjective t
+  simp only [toFun_eq_rTensor_φ_toFun' _ ha, smul_def, Pi.smul_apply, map_smul]
+
+end Module
+
+section ground
+
+variable {R : Type u} [CommSemiring R]
+    {M : Type*} [AddCommMonoid M] [Module R M]
+    {N : Type*} [AddCommMonoid N] [Module R N]
+    (f : M →ₚₗ[R] N)
+    (S : Type*) [CommSemiring S] [Algebra R S]
+
+theorem one_tmul_ground (x : M) :
+    1 ⊗ₜ f.ground x = f.toFun S (1 ⊗ₜ x) := by
+  simp only [ground, toFun'_eq_toFun]
+  convert f.isCompat_apply (Algebra.ofId R S) (1 ⊗ₜ[R] x)
+  · simp only [Function.comp_apply, TensorProduct.lid_symm_apply, TensorProduct.includeRight_lid]
+    congr
+  · rw [rTensor_tmul, toLinearMap_apply, _root_.map_one]
+
+end ground
+
+section Comp
+
+variable {R : Type u} [CommSemiring R]
+  {M : Type*} [AddCommMonoid M] [Module R M]
+  {N : Type*} [AddCommMonoid N] [Module R N]
+  {P : Type*} [AddCommMonoid P] [Module R P]
+  {Q : Type*} [AddCommMonoid Q] [Module R Q]
+  (f : M →ₚₗ[R] N) (g : N →ₚₗ[R] P) (h : P →ₚₗ[R] Q)
+
+/-- Extension of `MvPolynomial.comp_toFun'` -/
+theorem comp_toFun (S : Type*) [CommSemiring S] [Algebra R S] :
```

Comment:
Should this be `toFun_comp`? or `toFun_comp_eq_toFun_comp_toFun`?

Reply:
both are valid according to the naming convention but I think the first is more in line with the rest of the library.

Reply:
Can you explain me something, because now I'm even more confused than before?
I agreed that `toFun_add` reflects `toFun (f+g) S t = …`, which `add_toFun` does not. 
So why `toFun (f o g)` should be `toFun_comp`?
(I have no personal opinion, my goal is to understand this enough to fall naturally on the correct naming so as to avoid useless work in the review process.)